sklearnex/cluster/k_means.py

```diff
@@ -1,17 +1,370 @@
+    from sklearn.utils.validation import (
+        _check_sample_weight,
+        _deprecate_positional_args,
+        _num_samples,
+        check_is_fitted,
+    )
+
+    from daal4py.sklearn._device_offload import support_usm_ndarray
+    from daal4py.sklearn._n_jobs_support import control_n_jobs
+    from daal4py.sklearn._utils import sklearn_check_version
+    from onedal.cluster import KMeans as onedal_KMeans
+    from onedal.utils import _is_csr
+
+    from .._device_offload import dispatch, wrap_output_data
+    from .._utils import PatchingConditionsChain
+
+    class BaseKMeans(ABC):
+        def _get_cluster_centers(self):
+            return self._cluster_centers_
+
+        def _set_cluster_centers(self, value):
+            self._cluster_centers_ = value
+            if hasattr(self, "_onedal_estimator"):
+                self._onedal_estimator.cluster_centers_ = value
+
+        def _get_labels(self):
+            return self._labels_
+
+        def _set_labels(self, value):
+            self._labels_ = value
+            if hasattr(self, "_onedal_estimator"):
+                self._onedal_estimator.labels_ = value
+
+        def _get_inertia(self):
+            return self._inertia_
+
+        def _set_inertia(self, value):
+            self._inertia_ = value
+            if hasattr(self, "_onedal_estimator"):
+                self._onedal_estimator.inertia_ = value
+
+        def _get_n_iter(self):
+            return self._n_iter_
+
+        def _set_n_iter(self, value):
+            self._n_iter_ = value
+            if hasattr(self, "_onedal_estimator"):
+                self._onedal_estimator.n_iter_ = value
+
+        def _save_attributes(self):
+            assert hasattr(self, "_onedal_estimator")
+            self.n_features_in_ = self._onedal_estimator.n_features_in_
+            self.fit_status_ = 0
+            self._tol = self._onedal_estimator._tol
+            self._n_init = self._onedal_estimator._n_init
+            self._n_iter_ = self._onedal_estimator.n_iter_
+            self._labels_ = self._onedal_estimator.labels_
+            self._inertia_ = self._onedal_estimator.inertia_
+            self._algorithm = self._onedal_estimator.algorithm
+            self._cluster_centers_ = self._onedal_estimator.cluster_centers_
+            self._sparse = False
+
+            self.n_iter_ = property(self._get_n_iter, self._set_n_iter)
+            self.labels_ = property(self._get_labels, self._set_labels)
+            self.inertia_ = property(self._get_labels, self._set_inertia)
+            self.cluster_centers_ = property(
+                self._get_cluster_centers, self._set_cluster_centers
+            )
+
+            self._is_in_fit = True
+            self.n_iter_ = self._n_iter_
+            self.labels_ = self._labels_
+            self.inertia_ = self._inertia_
+            self.cluster_centers_ = self._cluster_centers_
+            self._is_in_fit = False
```

Comment:
Please change the creation of the property to follow this convention: https://github.com/intel/scikit-learn-intelex/pull/1843/files#r1652129017 I don't think you'll need to delete the onedal_model like in that PR, but follow the getter/setter strategy.  The method in this pollutes the class with methods more than necessary.